### DISCUSSION TITLE
Pushing different resolution for the same video

### DISCUSSION THREAD

**Discussion by lucabergamini** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#1088] [General]
Hello, this might be off-topic for this repo, so please feel free to point me to another one from awslab if that is the case.

We have a device pushing a video to kinesis streams. From time to time the device changes the resolution of the video it pushes (to save bandwidth and for other reasons), while keeping the same kinesis stream resource.

Everything works fine in the kinesis console in chrome (we can see the resolution switching and the video playing, the resolution is correctly reported) and we can serve it fine in all our hls.js clients.

However, for ios safari (where we are forced to use the native html player), safari on macos (when using the native html player, given that we can choose there) and even quicktime the video hangs at the resolution switch. This happens in the kinesis video player too on safari.

There are no errors and the player actually thinks it's still playing, but the image is still. One hypothesis is that Apple is interpreting the specs https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-2, in a different way and it's expecting the playlist (we use a single playlist, but fragments can change resolution) to use a single resolution, or maybe we are not providing some metadata while pushing the data to kinesis.

Is our approach of pushing to the same kinesis stream fragments with a different resolution sensible, or should we use a different approach?

We have opened an issue in webkit, but it would be great to at least confirm we are doing the right thing from the hls perspective.

Cheers,
Luca

**Comment by niyatim23** [comment]
Hi @lucabergamini, that is the correct way to do things. This is not an issue with the SDK but could probably be an issue with the playback of the video on Apple's side or with KVS playback. We are actively tracking this item internally to figure out the disconnect with Apple. We will provide you with an update once we have something
